apps/web/tests/asset-inventory/collector/collector-details/edit-collector.spec.ts

```diff
@@ -0,0 +1,295 @@
+            // Save changes.
+            await saveChanges(0);
+            try {
+                const apiResponse = await page.waitForResponse(`${process.env.APIURL as string}/inventory/collector/update`);
+                await page.waitForLoadState('networkidle');
+                expect(apiResponse.status()).toBe(200);
+            } catch (error) {
+                console.error('Error occurred:', error);
+            }
+
+            // Check if the tags are added.
+            const tags = page.locator('.tags-wrapper .p-tag:first-child');
+            const keyElement = await tags.locator('.key');
+            const valueElement = await tags.locator('.value');
+            await expect(keyElement).toContainText(randomKey);
+            await expect(valueElement).toContainText(randomValue);
+        });
+
+        const firstTimeBlock = page.locator('.collector-schedule-edit-form .hourly-schedule-wrapper .time-block').nth(0);
+
+        await test.step('7-9. Edit schedule and save changes', async () => {
+            // Click edit button.
+            const editButton = page.locator('.collector-detail-page .section').nth(1).locator('.extra .p-button');
+            await editButton.first().click();
+
+            const scheduleToggleButton = page.locator('.collector-schedule-edit-form .p-toggle-button');
+            const toggleLabel = scheduleToggleButton.locator('.state-text');
+
+            // Check if the toggle is in the active state
+            const toggleClassAttribute = await scheduleToggleButton.getAttribute('class');
+            const hasToggleActiveClass = toggleClassAttribute.includes('is-active');
+
+            if (hasToggleActiveClass) {
+                await expect(toggleLabel).toContainText('ON');
+
+                // Select the time block.
+                const secondTimeBlock = page.locator('.collector-schedule-edit-form .hourly-schedule-wrapper .time-block').nth(1);
+                await firstTimeBlock.first().click();
+                await secondTimeBlock.first().click();
+
+                // Save changes.
+                await saveChanges(1);
+                try {
+                    const apiResponse = await page.waitForResponse(`${process.env.APIURL as string}/inventory/collector/update`);
+                    await page.waitForLoadState('networkidle');
+                    expect(apiResponse.status()).toBe(200);
+                } catch (error) {
+                    console.error('Error occurred:', error);
+                }
+
+                // Check if the schedule is set.
+                const classAttribute = await firstTimeBlock.getAttribute('class');
+                const hasSetScheduleClass = classAttribute.includes('active');
+                if (hasSetScheduleClass) {
+                    await expect(hasSetScheduleClass).toBeTruthy();
+                } else {
+                    console.log('Test skipped due to no selected schedule.');
+                }
```

Comment:
I excluded it because it checks along with the first time block and I thought the 'active' status would have been changed similarly. However, I will add a case for more accurate testing.